File: Programming/JavaScript/Maximum-Gap/MaximumGap.js
function maximumGap(arr) {
	let max = 0;
	if (arr.length >= 2) {
		arr.sort((a, b) => a - b);

		for (let i in arr) {
			const diff = arr[i] - arr[i - 1];
			
			if (diff > max) {
				max = diff;
			}
		}

	}
	return max;
}

const arr = [1, 30, 4, 21, 10];
console.log(maximumGap(arr));

File: Programming/C#/MaximumGap/MaximumGap.cs
using System;

namespace Maximum_Gap
{
    class MaximumGap
    {
        static void Main(string[] args)
        {
            int arraySize = 0;

            Console.Write("Number of elements in the array: ");
            arraySize = Convert.ToInt32(Console.ReadLine());
            Console.WriteLine();

            int[] array = new int[arraySize];

            for(int i = 0; i < arraySize; i++)
            {
                Console.Write("Element #{0}: ", i+1);
                array[i] = Convert.ToInt32(Console.ReadLine());
            }

            for(int i = 0; i < arraySize; i++)
                for (int j = 1; j < arraySize; j++)
                    if(array[i] > array[j])
                    {
                        int aux = array[i];
                        array[i] = array[j];
                        array[j] = aux;
                    }

            int maxGap = 0;

            if(arraySize >= 2) {
            for (int i = 0; i < arraySize - 1; i++)
                if (array[i] - array[i + 1] > maxGap)
                    maxGap = array[i] - array[i + 1];
            }

            Console.WriteLine();
            Console.WriteLine("Max gap: {0}", maxGap);

            Console.WriteLine();
            Console.WriteLine("Press enter to close...");
            Console.ReadLine();
        }
    }
}
